apps/outlook-rooms-addin/src/tests/filter-space.component.spec.ts

```diff
@@ -0,0 +1,199 @@
+                useValue: {
+                    dismiss: jest.fn(),
+                },
+            },
+            {
+                provide: FeaturesFilterService,
+                useValue: {
+                    features$: of(mockFeatures),
+                    applyFilter: jest.fn(),
+                    getSelectedFeatures: jest.fn(),
+                },
+            },
+            {
+                provide: EventFormService,
+                useValue: {
+                    setView: jest.fn(() => {}),
+                    newForm: jest.fn(() => {}),
+                    clearForm: jest.fn(),
+                    storeForm: jest.fn(() => {}),
+                    loadForm: jest.fn(),
+                    postForm: jest.fn(),
+                    setOptions: jest.fn(() => {}),
+                    view: '',
+                    available_spaces: of(
+                        mockSpace,
+                        mockSpace,
+                        mockSpace,
+                        mockSpaceWithViews
+                    ),
+                    options: of(mockEventFlowOptions),
+                },
+            },
+        ],
+        declarations: [],
+    });
+
+    beforeEach(() => {
+        spectator = createComponent();
+        jest.clearAllMocks();
+        jest.resetModules();
+        ngMocks.reset();
+        spectator = createComponent();
+        spectator.inject(FeaturesFilterService);
+        const event_service: any = spectator.inject(EventFormService);
+        event_service.setOptions.mockImplementation(() => {
+            return of(mockEventFlowOptions);
+        });
+
+        event_service.setView.mockImplementation((_) => {
+            event_service.view = _;
+            spectator.detectChanges();
+        });
+        event_service.newForm.mockImplementation((_) => {
+            event_service.form = fb.group(formModel);
+        });
+        event_service.storeForm.mockImplementation((_) => {
+            if (
+                event_service.form.controls.title == null ||
+                event_service.form.controls.date == null ||
+                event_service.form.controls.duration == null
+            )
+                return null;
+        });
+    });
+
+    it('should create component', () => {
+        expect(spectator.component).toBeTruthy();
+    });
+    it('should update features based on checkboxes', async () => {
+        const filter_service = spectator.inject(FeaturesFilterService);
+        const event_service = spectator.inject(EventFormService);
+        await event_service.newForm();
+
+        const get_features_spy = jest.spyOn(
+            filter_service,
+            'getSelectedFeatures'
+        );
+        const update_features_spy = jest.spyOn(filter_service, 'applyFilter');
+        const check_boxes = spectator.debugElement.queryAll(
+            By.css('mat-checkbox')
+        );
+
+        spectator.component.ngOnInit();
+        spectator.detectChanges();
```

Comment:
Unless you need to check the template for changes you shouldn't need to call `detectChanges`.